core/src/main/java/io/fabric8/maven/core/config/Profile.java

```diff
@@ -56,4 +102,30 @@ public ProcessorConfig getGeneratorConfig() {
         return generatorConfig;
     }
 
+    public int getOrder() {
+        return order;
+    }
+
+    @Override
+    public int compareTo(Profile o) {
+        int orderDiff = order - o.order;
+        if (orderDiff != 0) {
+            return orderDiff;
+        }
+        // Equals is only the very same object. This is ok, since at the end
+        // profiles are merge to a single profile
+        if (this.equals(o)) {
+            return 0;
+        } else {
+            // Try to determine the order based on creation time. The rule is, a newer profile
+            // is supposed to have a higher order.
+            long timeDiff = creationTime - o.creationTime;
```

Comment:
Not really since these profiles which are 'sorted' come from a `ClassLoader.getResources()` which is also non deterministic. I don't mind to have the order of Profiles unspecified when the order number is the same. However it should be stable. 

Actually the sorting seems to be still a bit messy since it happens separately in different list while collecting profiles and not at the end in an overall list. So, I need to work on this anyway still.

tbh, don't know whether this is really so important. however I think a consistent behaviour is important.

So I suggest that we use an static ordinal counter (AtomicInteger) and count that up and use this for the discriminator. Guess this makes most sense.